nilearn/plotting/surface/surf_plotting.py

```diff
@@ -0,0 +1,938 @@
+            This option is currently only implemented for the
+            ``matplotlib`` engine.
+
+    %(cbar_tick_format)s
+        Default="auto" which will select:
+
+        - '%%.2g' (scientific notation) with ``matplotlib`` engine.
+        - '.1f' (rounded floats) with ``plotly`` engine.
+
+        .. versionadded:: 0.7.1
+
+    %(title)s
+
+    title_font_size : :obj:`int`, default=18
+        Size of the title font.
+
+        .. note::
+            This option is currently only implemented for the
+            ``plotly`` engine.
+
+        .. versionadded:: 0.9.0
+
+    %(output_file)s
+
+    axes : instance of matplotlib axes or None, default=None
+        The axes instance to plot to. The projection must be '3d' (e.g.,
+        `figure, axes = plt.subplots(subplot_kw={'projection': '3d'})`,
+        where axes should be passed.).
+        If None, a new axes is created.
+
+        .. note::
+            This option is currently only implemented for the
+            ``matplotlib`` engine.
+
+    %(figure)s
+
+        .. note::
+            This option is currently only implemented for the
+            ``matplotlib`` engine.
+
+    Returns
+    -------
+    fig : :class:`~matplotlib.figure.Figure` or\
+    :class:`~nilearn.plotting.displays.PlotlySurfaceFigure`
+        The surface figure. If ``engine='matplotlib'`` then a
+        :class:`~matplotlib.figure.Figure` is returned.
+        If ``engine='plotly'``, then a
+        :class:`~nilearn.plotting.displays.PlotlySurfaceFigure`
+        is returned
+
+    See Also
+    --------
+    nilearn.datasets.fetch_surf_fsaverage : For surface data object to be
+        used as background map for this plotting function.
+
+    nilearn.plotting.plot_surf_roi : For plotting statistical maps on brain
+        surfaces.
+
+    nilearn.plotting.plot_surf_stat_map : for plotting statistical maps on
+        brain surfaces.
+
+    nilearn.surface.vol_to_surf : For info on the generation of surfaces.
+    """
+    fig = get_surface_backend(engine).plot_surf(
+        surf_mesh=surf_mesh,
+        surf_map=surf_map,
+        bg_map=bg_map,
+        hemi=hemi,
+        view=view,
+        cmap=cmap,
+        symmetric_cmap=symmetric_cmap,
+        colorbar=colorbar,
+        avg_method=avg_method,
+        threshold=threshold,
+        alpha=alpha,
+        bg_on_data=bg_on_data,
+        darkness=darkness,
+        vmin=vmin,
+        vmax=vmax,
+        cbar_vmin=cbar_vmin,
+        cbar_vmax=cbar_vmax,
+        cbar_tick_format=cbar_tick_format,
+        title=title,
+        title_font_size=title_font_size,
+        output_file=output_file,
+        axes=axes,
+        figure=figure,
+    )
+
+    return fig
+
+
+@fill_doc
+def plot_surf_contours(
+    surf_mesh=None,
+    roi_map=None,
+    hemi=None,
+    levels=None,
+    labels=None,
+    colors=None,
+    legend=False,
+    cmap="tab20",
+    title=None,
+    output_file=None,
+    axes=None,
+    figure=None,
+    **kwargs,
+):
```

Comment:
This is not implemented for plotly. But I've put a method in plotly backend and raise NotImplementedError. I thought of adding `engine` parameter here. Would that be ok?